src/Endatix.Api/Endpoints/Auth/Register.cs

```diff
@@ -24,18 +27,15 @@ public override void Configure()
         });
     }
 
-    public override async Task HandleAsync(RegisterRequest request, CancellationToken cancellationToken)
+    /// <inheritdoc/>
+    public override async Task<Results<Ok<RegisterResponse>, BadRequest<Errors.ProblemDetails>>> ExecuteAsync(RegisterRequest request, CancellationToken cancellationToken)
     {
         var registerUserCommand = new RegisterCommand(request.Email, request.Password);
-        var result = await mediator.Send(registerUserCommand, cancellationToken);
-
-        if (!result.IsSuccess)
-        {
-            ThrowError("Registration failed. Please check your input and try again.");
-        }
-
-        var successfulResponse = new RegisterResponse(Success: true, Message: "User has been successfully registered");
+        var userRegistrationResult = await mediator.Send(registerUserCommand, cancellationToken);
 
-        await SendOkAsync(successfulResponse, cancellationToken);
+        return TypedResultsBuilder
+                .MapResult(userRegistrationResult, (user) => new RegisterResponse(Success: true, Message: "User has been successfully registered"))
+                .SetErrorMessage("Registration failed. Please check your input and try again.")
```

Comment:
I agree. Magic strings are bad code smell.

This was the state before my refactoring though, so it's an inherited problem in this class. Honestly, I decided not to do it as I thought this will be addressed once we move to proper support for ProblemDetails. Ideally, the message should be a public prop, so it can be used in test assertions as well. 

Let's address this as part of this task - [feat: EndatixAPI: Implement ProblemDetails Response](https://github.com/endatix/endatix/issues/168)
#168. I have added a reference to the comment. As part of this task Register will need to be refactored anyways, so will do it with it. Agree?

Reply:
Sounds good, thank you.